lib/spack/env/cc

```diff
@@ -271,75 +339,64 @@ for dep in "${deps[@]}"; do
     # Prepend include directories
```

Comment:
It looks like you are appending now? Is this safe to do since for any conflicts it will now it will pick the ones passed as args to the wrapper instead of the explicit dependences? If appending is really what we want, then please modify the "prepend" comments.